### ISSUE TITLE
Redirect URI's incorrectly forced as unique

### ISSUE THREAD

**Issue by KimBurgess** [PlaceOS/models#17]
The current [implementation for app id generation](https://github.com/PlaceOS/models/blob/e90c34dc5747fd9ceed6913767b36cd30df2c8fd/src/models/doorkeeper_application.cr#L34-L36) is based off an MD5 of the redirect URI. This prevents the same URI from being used on multiple app registrations.

Given that apps now bind to authorities (domains) as the owner, there are use cases where having multiple apps sharing a redirect URI are required. An example of this would be a native app registered for access across multiple auth domains.

**Comment by stakach** [comment]
The web apps currently use this structure so the same static set of files can be used across multiple domains. So we could just check if it's a HTTP(S) protocol and use a UUID if it's not?

**Comment by KimBurgess** [comment]
Clever solution! There are scenarios where who may want multiple registrations for web apps too, such as different scopes if/when support is expanded for that.

Maybe a fix at the model level is to just default to that digest, so that it can be explicitly defined if required upstream.
```crystal
self.uid ||= Digest::MD5.hexdigest(self.redirect_uri.not_nil!)
```

**Comment by stakach** [comment]
I can do both

**Comment by stakach** [comment]
fixed in #20
